File: 백준/Gold/1253. 좋다/좋다.cc
#include<iostream>
#include<algorithm>
using namespace std;

int arr[2005];

int main(void) {
	cin.tie(NULL)->sync_with_stdio(false);

	int N,target;
	int result = 0;

	cin >> N;
	for (int n = 0; n < N; ++n) {
		cin >> arr[n];
	}

	sort(arr, arr + N);

	for (int i = 0; i < N; ++i) {
		target = arr[i];
		int st = 0, en = N - 1, sum;
		while (st < en) {
			sum = arr[st] + arr[en];
			if (sum == target) {
				if (st != i && en != i) {
					result++;
					break;
				}
				else if (st == i) st++;
				else if (en == i) en--;
			}
			else if (sum < target) st++;
			else en--;
		}
	}

	cout << result<<'\n';

	return 0;
}

File: 백준/Gold/1253. 좋다/좋다.java
import java.io.*;
import java.util.*;

public class Main {
    static int N;
    static int arr[];

    public static void main(String args[]) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        N = Integer.parseInt(br.readLine());
        arr = new int[N];
        StringTokenizer st = new StringTokenizer(br.readLine());
        for (int i = 0; i < N; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }

        Arrays.sort(arr);

        int ans = 0;
        for (int i = 0; i < N; i++) {
            int cur = arr[i];
            int left = 0, right = N - 1;
            boolean isGood = false;
            while (left < right) {
                int sum = arr[left] + arr[right];
                if (sum == cur) {
                    if (left != i && right != i) {
                        ans++;
                        break;
                    } else if (left == i) left++;
                    else if (right == i) right--;
                }
                else if (sum < cur) left++;
                else right--;
            }
        }
        System.out.println(ans);
    }
}
